load-tests/proto3_test/src/server.bal

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/serdes;
+import ballerina/io;
+import ballerina/http;
+import ballerina/time;
+
+configurable int PORT = 9100;
+const time:Seconds EXECUTION_TIME = 10;
+
+time:Utc startedTime = time:utcNow();
+time:Utc endedTime = time:utcNow();
+
+isolated int errorCount = 0;
+isolated int serializationCount = 0;
+isolated int deserializationCount = 0;
+
+boolean completed = false;
+
+type User record {
+    readonly int id;
+    string name;
+    int age;
+};
+
+table<User> key(id) users = table [];
+
+service /serdes on new http:Listener(PORT) {
+
+    function init() returns error? {
+        check loadUserTable();
+    }
+
+    resource function get 'start() returns boolean {
+        http:Client|error clientEP = new (string `http://localhost:${PORT}/serdes`);
+        if clientEP is error {
+            return false;
+        }
+        resetCountersAndTime();
+        _ = start beginCommunication(clientEP);
+        return true;
+    }
+
+    resource function post next(http:Request request) returns byte[]|error {
+        serdes:Proto3Schema serdes = check new (User);
+        byte[] encodedPayload = check request.getBinaryPayload();
+        User nextUser = handleDeserialization(serdes, encodedPayload);
+        return handleSerialization(serdes, nextUser);
+    }
+
+    resource function get result() returns boolean|map<string> {
```

Comment:
Better if we can return nil here instead of a boolean since we only return false in the boolean scenario.

Reply:
updated